packages/chakra-ui-core/src/CButton/CButton.js

```diff
@@ -77,63 +74,50 @@ const CButtonIcon = {
  *
  * The Button component is an accessible rich component that does what a button does :)
  *
- * @extends CPseudoBox
  * @see Docs https://vue.chakra-ui.com/button
  */
 const CButton = {
-  name: 'CButton',
-  inject: ['$chakraTheme', '$chakraColorMode'],
-  props: {
-    ...buttonProps,
-    ...styleProps,
-    to: [String, Object]
-  },
+  mixins: [createStyledAttrsMixin('CButton', true)],
+  props: buttonProps,
   computed: {
     colorMode () {
       return this.$chakraColorMode()
     },
     theme () {
       return this.$chakraTheme()
+    },
+    componentStyles () {
+      return createButtonStyles({
+        color: this.variantColor,
+        variant: this.variant,
+        theme: this.theme,
+        ripple: this.ripple,
+        colorMode: this.colorMode,
+        size: this.size || 'md'
+      })
     }
   },
   render (h) {
-    const buttonStyles = createButtonStyles({
-      color: this.variantColor,
-      variant: this.variant,
-      theme: this.theme,
-      ripple: this.ripple,
-      colorMode: this.colorMode,
-      size: this.size || 'md'
-    })
-
-    return h(CPseudoBox, {
-      props: {
-        as: this.as,
-        to: this.to,
-        outline: 'none',
-        cursor: 'pointer',
-        fontSize: 'md',
-        fontWeight: '700',
-        border: 'none',
-        rounded: 'md',
-        width: this.isFullWidth ? 'full' : undefined,
-        ...buttonStyles,
-        ...forwardProps(this.$props)
-      },
+    return h(this.as, {
+      class: this.className,
       attrs: {
         type: this.type,
         tabIndex: 0,
         disabled: this.isDisabled || this.isLoading,
         'aria-disabled': this.isDisabled || this.isLoading,
         dataActive: this.isActive ? 'true' : undefined,
-        'data-chakra-component': 'CButton'
+        'data-chakra-component': 'CButton',
```

Comment:
`data-chakra-component` this is not necessary here

Reply:
Good point. Forgot to remove this.